doc/extdev/howto.rst

```diff
@@ -0,0 +1,170 @@
+Extension How-tos
```

Comment:
I'd like to see this document split up. I think there are at least two separate documents here: one related to development of themes for the HTML builder (for the custom template functions and injection of Javascript sections), a builders document (for the entry point discussion), and a general overview document (for the inter-extension dependency documentation).

I'd also like to see these documents go into the `doc/development` directory as that's where everything in the `extdev` directory should be moved eventually.

Reply:
ok I'll give this a shot at splitting into 3 files